Incorrect Documentation about source of Error

https://github.com/nasa/cumulus/blob/3b07b1c0fba29be8d860041959f5d5deeb03f454/docs/system-documentation/system-documentation.md

The section at the bottom: **Error: Unable to import module 'index': Error**

And the solution: "In order to resolve this issue, update the lambda source (in lambdas.yml, to point to the parent directory of the index.js file."

This is not correct. Add **any** syntax error **inside** your index.js file, and CloudWatch Logs will report this error. The error means that it is unable to import the module [because of some error].

It is not an error referencing the file.


When I expand the section in AWS Step Functions, it reveals the real error with my index.js file:

```
"cause": {
    "errorMessage": "Cannot find module 'node_modules/@cumulus/cumulus-message-adapter-js'",
```


But clicking the button in the Step Function console that says "CloudWatch Logs" just shows me the generic:

`Unable to import module 'index': Error`

--- a/docs/system-documentation/system-documentation.md
+++ b/docs/system-documentation/system-documentation.md
@@ -30,7 +30,7 @@ Visual representations of executed workflows can be found in the Cumulus dashboa
 
 If a workflow errors, the error will be handled according to the [error handling configuration](data-cookbooks/error-handling.md). The task that fails will have the `exception` field populated in the output, giving information about the error. Further information can be found in the CloudWatch logs for the task.
 
-![](assets/workflow-fail.png)
+![Graph of AWS Step Function execution showing a failing workflow](assets/workflow-fail.png)
 
 ### Workflow Did Not Start
 
@@ -51,24 +51,27 @@ The above error was being thrown by cumulus lambda function invocation. The KMS 
 On a lambda level, this error can be resolved by updating the KMS Key to `aws/lambda`. We've done this through the management console. Unfortunately, this approach doesn't scale well.
 
 The other resolution (that scales but takes some time) that was found is as follows:
+
 1. Delete the whole `{{#each newsted_templates}}` section from `@cumulus/deployment/app/cloudformation.template.yml` and redeploy the primary stack.
 2. Reinstall dependencies via `npm`.
 3. Re-deploy the stack.
 
-
 [Discussed in the Earthdata Wiki](https://wiki.earthdata.nasa.gov/display/CUMULUS/KMS+Exception%3A+AccessDeniedException).
-
 
 ### Error: Unable to import module 'index': Error
 
-This error is shown in the CloudWatch logs for a lambda function. The cause is a lambda defined in `lambdas.yml` that is pointing to an `index.js` source file. In order to resolve this issue, update the lambda source (in `lambdas.yml`, to point to the parent directory of the `index.js` file.
+This error is shown in the CloudWatch logs for a Lambda function.
 
-```
+One possible cause is that the Lambda definition in `lambdas.yml` is not pointing to the directory for the `index.js` source file. In order to resolve this issue, update the lambda definition in `lambdas.yml` to point to the parent directory of the `index.js` file.
+
+```yaml
 DiscoverGranules:
   handler: index.handler
   timeout: 300
   source: 'node_modules/@cumulus/discover-granules/dist/'
   useMessageAdapter: true
 ```
+
+If you are seeing this error when using the Lambda as a step in a Cumulus workflow, then inspect the output for this Lambda step in the AWS Step Function console. If you see the error `Cannot find module 'node_modules/@cumulus/cumulus-message-adapter-js'`, then you need to set `useMessageAdapter: true` in the Lambda definition in `lambdas.yml`.
 
 [Discussed in the Earthdata Wiki](https://wiki.earthdata.nasa.gov/display/CUMULUS/Troubleshooting).

--- a/docs/workflows/developing-workflow-tasks.md
+++ b/docs/workflows/developing-workflow-tasks.md
@@ -5,6 +5,7 @@ hide_title: true
 ---
 
 # Developing Workflow Tasks
+
 Workflow tasks can be either AWS Lambda Functions or ECS Activities.
 
 ## Lambda functions

--- a/docs/workflows/lambda.md
+++ b/docs/workflows/lambda.md
@@ -6,19 +6,23 @@ hide_title: true
 
 # Develop Lambda Functions
 
-### Develop a new Lambda
+## Develop a new Lambda
 
 To develop a new lambda from a sample, create a new folder in `cumulus/tasks/` and run `npm init`:
 
-    $ cd ../cumulus/tasks
-    $ mkdir new-lambda
-    $ cd new-lambda
-    $ npm init
+```bash
+  $ cd ../cumulus/tasks
+  $ mkdir new-lambda
+  $ cd new-lambda
+  $ npm init
+```
 
 Or copy an existing lambda function to customize:
 
-    $ cd ../cumulus/tasks
-    $ cp discover-pdrs new-lambda
+```bash
+  $ cd ../cumulus/tasks
+  $ cp discover-pdrs new-lambda
+```
 
 Modify package.json:
 
@@ -28,42 +32,54 @@ Modify package.json:
 * test script
 * dependencies (NOT devDependencies)
 
-### Build a Lambda
+## Build a Lambda
 
-To build node.js lambda functions, use webpack to pack into single .js with dependencies:
+To build Node.js lambda functions, use webpack to pack the Lambda code into a single `.js` file with dependencies:
 
-    $ npm run build
+```bash
+  $ npm run build
+```
 
 Alternatively, to monitor for changes and auto-rebuild:
 
-    $ npm run watch
+```bash
+  $ npm run watch
+```
 
 For non-node lambdas not included in Cumulus repo, upload .zip to s3 and modify lambdas.yml as previously shown.
 
-### Deploy a Lambda
+## Deploy a Lambda
 
-For new lambdas, update `<daac>-deploy/lambdas.yml` by adding a new entry.
-E.g.: node.js sample for '../cumulus/cumulus/tasks/sample-lambda' in the Cumulus repo):
+For new Node.js lambdas, update `<daac>-deploy/lambdas.yml` by adding a new entry.
 
-    <LambdaName>:                                       # eg:  LambdaSample (does not need to conform to dirname)
-      handler: <dir>.<function>                                # eg:  sample-lambda.handler (assuming file has module.exports.handler = <someFunc>)
-      timeout: <s>                                             # eg:  300
-      source: 'node_modules/@cumulus/<dir>/dist/'  # eg:  '../cumulus/cumulus/tasks/sample-lambda/dist/index.js'
+```yaml
+    <LambdaName>:                                   # eg:  LambdaSample (does not need to conform to dirname)
+      handler: <dir>.<function>                     # eg:  sample-lambda.handler (assuming file has module.exports.handler = <someFunc>)
+      timeout: <s>                                  # eg:  300
+      source: 'node_modules/@cumulus/<dir>/dist/'   # eg:  '../cumulus/cumulus/tasks/sample-lambda/dist/index.js'
+      useMessageAdapter: true                       # necessary if this Lambda is included as part of a Cumulus workflow
+```
 
-For non-node.js lambda code (e.g. python) uploaded as a .zip to an S3 bucket:
+For non-Node.js lambda code (e.g. python) uploaded as a .zip to an S3 bucket:
 
-    PyLambda:
-      handler: <file.py>.<function>               # eg:  lambda_handler.handler for lambda_handler.py with:  def handler(event, context):
-      timeout: <s>
-      s3Source:
-        bucket: '{{buckets.internal.name}}'       # refers to bucket set in config.yml
-        key: deploy/cumulus-process/<dir>/<file>  # eg: deploy/cumulus-process/modis/0.3.0b3.zip
-      runtime: python2.7                          # Node is default, otherwise specify.
+```yaml
+  PyLambda:
+    handler: <file.py>.<function>               # eg:  lambda_handler.handler for lambda_handler.py with:  def handler(event, context):
+    timeout: <s>
+    s3Source:
+      bucket: '{{buckets.internal.name}}'       # refers to bucket set in config.yml
+      key: deploy/cumulus-process/<dir>/<file>  # eg: deploy/cumulus-process/modis/0.3.0b3.zip
+    runtime: python2.7                          # Node is default, otherwise specify.
+```
 
-To deploy all changes to /tasks/ and lambdas.yml:
+To deploy all changes to `/tasks/` and `lambdas.yml`:
 
-    $ kes cf deploy --kes-folder app --template ../cumulus/packages/deployment/app --region <region> --deployment <deployment-name> --role <arn:deployerRole>
+```bash
+  $ kes cf deploy --kes-folder app --template ../cumulus/packages/deployment/app --region <region> --deployment <deployment-name> --role <arn:deployerRole>
+```
 
 To deploy modifications to a single lambda package:
 
-    $ kes lambda <LambdaName> --kes-folder app --template ../cumulus/packages/deployment/app --deployment <deployment-name> --role <arn:deployerRole>
+```bash
+  $ kes lambda <LambdaName> --kes-folder app --template ../cumulus/packages/deployment/app --deployment <deployment-name> --role <arn:deployerRole>
+```

